internal/services/automation/automation_webhook_test.go

```diff
@@ -0,0 +1,255 @@
+				check.That(data.ResourceName).Key("name").Exists(),
+				check.That(data.ResourceName).Key("resource_group_name").Exists(),
+				check.That(data.ResourceName).Key("automation_account_name").Exists(),
+				check.That(data.ResourceName).Key("expiry_time").Exists(),
+				check.That(data.ResourceName).Key("enabled").HasValue("true"),
+				check.That(data.ResourceName).Key("runbook_name").HasValue("Get-AzureVMTutorial"),
+				check.That(data.ResourceName).Key("parameters").DoesNotExist(),
+				check.That(data.ResourceName).Key("uri").Exists(),
+			),
+		},
+		data.ImportStep("uri"),
+	})
+}
+
+func TestAccAutomationWebhook_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_automation_webhook", "test")
+	r := AutomationWebhookResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.SimpleWebhook(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config:      r.requiresImport(data),
+			ExpectError: acceptance.RequiresImportError("azurerm_automation_webhook"),
+		},
+	})
+}
+
+func TestAccAutomationWebhook_WithParameters(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_automation_webhook", "test")
+	r := AutomationWebhookResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.WebhookWithParameters(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("parameters.input").HasValue("parameter"),
+				check.That(data.ResourceName).Key("uri").Exists(),
+			),
+		},
+		data.ImportStep("uri"),
+	})
+}
+
+func TestAccAutomationWebhook_ChangeUri(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_automation_webhook", "test")
+	r := AutomationWebhookResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.WebhookURIChange(data, "https://12345678-9012-3456-7890-123456789012.webhook.we.azure-automation.net/webhooks?token=abcdefghijklmnoprstuwxyz1234567890abcdefghijklm"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("uri").HasValue("https://12345678-9012-3456-7890-123456789012.webhook.we.azure-automation.net/webhooks?token=abcdefghijklmnoprstuwxyz1234567890abcdefghijklm"),
+			),
+		},
+		data.ImportStep("uri"),
+		{
+			Config: r.WebhookURIChange(data, "https://12345678-9012-3456-7890-123456789012.webhook.we.azure-automation.net/webhooks?token=abcdefghijklmnoprstuwxyz1234567890abcdefg313377"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("uri").HasValue("https://12345678-9012-3456-7890-123456789012.webhook.we.azure-automation.net/webhooks?token=abcdefghijklmnoprstuwxyz1234567890abcdefg313377"),
+			),
+		},
+	})
+}
+
+/*
+// missing workergroup creation
```

Comment:
is there no way to create a worker group yet?

Reply:
It's not exposed in [Azure REST API](https://docs.microsoft.com/en-us/rest/api/automation/hybrid-runbook-worker-group). I guess that this might be due to the fact, that worker group is created when agent is installed and registered in Azure. I assumed, that this is only resource missing in terraform, but as I've just now checked, I think it is better just to remove this case

Reply:
can you pass in a fake value?

Reply:
I can pass the fake value, the only thing is, that there will be an error returned from REST API. As it is quite specific, I've implemented it  right now, as expecting this specific error.